src/server/project.ts

```diff
@@ -928,19 +970,27 @@ namespace ts.server {
         }
     }
 
+    type WildCardDirectoryWatchers = { watcher: FileWatcher, recursive: boolean };

```

Comment:
Nit: Use an interface instead of a type literal.
Also, it seems inconsistent that we use `recursive: boolean` here but `WatchDirectoryFlags` elsewhere.